src/index.js

```diff
@@ -33,42 +37,98 @@ function fetchData(searchTerm) {
         .catch(function (error) { console.log(`${error.name}:\n${error.message}`); });
 }
 
-function decodeData(foundData) {
-    for (let foundArticle of foundData.query.search) {
-        // console.log(foundArticle.snippet)
 
+async function decodeData(foundData) {
+
+    prepareResultsSection()     // create section in html where resutls will be displayed
+
+    for (let foundArticle of foundData.query.search) {
+        // console.log(foundData)
         let articleTitle = foundArticle.title
+
         let articleText = foundArticle.snippet
-        .replace(/(<([^>]+)>)/gi, "")               // strip html tags from 
-        .slice(0, 130) + "...";                      // truncate text
-        
-        let articleUrl = 'https://en.wikipedia.org/?curid=' + Number(foundArticle.pageid)        //
+            .replace(/(<([^>]+)>)/gi, "")               // strip html tags from 
+            .slice(0, 130) + "...";                      // truncate text
+
+        let pageID = Number(foundArticle.pageid)
 
-        generateCard(articleTitle, articleText, articleUrl)
+        let articleUrl = 'https://en.wikipedia.org/?curid=' + pageID
+
+        let imgUrl = await getArticleImage(articleTitle, pageID)
+        // console.log(imgUrl)
+
+        generateCard(articleTitle, articleText, articleUrl, imgUrl)
     }
 }
 
+async function getArticleImage(title, pageID) {
+    let url = baseUrl + "?origin=*" + `&action=query&format=json&prop=pageimages&titles=${title}&pithumbsize=260`;        // "?origin=*" to resolve cors cross-origin request issues
+    let imgURL
 
-function generateCard(title, text, url) {
+    let res = await fetch(url)
+    let data = await res.json()
+    try {
+        imgURL = data.query.pages[String(pageID)].thumbnail.source
+        // console.log(data.query.pages[String(pageID)].thumbnail.source)
+    } catch (err) {
+        imgURL = '../img/default.jpg'
+    }
+
+    // console.log(imgURL)
+    return imgURL
+}
+
+function prepareResultsSection() {
+    resultsSection.textContent = ''
+    let containerDiv = document.createElement('div')
+    let rowDiv = document.createElement('div')
+    let colDiv = document.createElement('div')
+    let h2 = document.createElement('h2')
+
+    containerDiv.className = "container"
+    rowDiv.className = "row"
+    rowDiv.id = "found"
+    colDiv.className = "col-12"
+    h2.classList.add("mb-3", "text-danger")
+    h2.textContent = "Your results: "
+    colDiv.appendChild(h2)
+    rowDiv.appendChild(colDiv)
+    containerDiv.appendChild(rowDiv)
+    resultsSection.appendChild(containerDiv)
+    /*
+        resultsSection.innerHTML = `
+        <div class="container">
+            <div class="row" id="found">
+                <div class="col-12">
+                    <h2 class="mb-3 text-danger">Your results: </h2>
+                </div>
+            </div>
+        </div>
+        `
+    */
+}
```

Comment:
I love how you write this. 
1. Have a final HTML code outcome as a comment
2. create all element in one part
3. gives attribute to created elements
4. append them at the end

Nice work